osu.Game/Overlays/BeatmapSet/BeatmapBadgePill.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.LocalisationExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Localisation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+#nullable enable
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public abstract class BeatmapBadgePill : CompositeDrawable
```

Comment:
bit of a nit but this is weird naming - is this a "badge" or a "pill"? why both?

Reply:
Wasn't sure if the term `Badge` would be enough to indicate a drawable, so added a `Pill` at the end. Could trim the latter and rename all subclasses accordingly.

Reply:
"badge" is already an established term for this sort of component in [probably the most popular website styling framework out there](https://getbootstrap.com/docs/5.2/components/badge/). should be safe to drop the "pill" thing.